### PR TITLE
 Generate AccessContextManager AccessLevel in Terraform.

### PR DESCRIPTION
**Pull request by rileykarson**
This API has some quirks - namely that `name`, the `self_link` equivalent is an input & redundant with the `parent` variable we use for `Create` operations.

-----------------------------------------------------------------
# [all]
### [terraform-beta]
Add support for access_context_manager_access_level

### CODE REVIEW

File: products/accesscontextmanager/api.yaml [hunk 1]
```diff
@@ -66,3 +66,200 @@ objects:
         description: |
           Time the AccessPolicy was updated in UTC.
         output: true
+  - !ruby/object:Api::Resource
+    name: 'AccessLevel'
+    # This is a weird API, so we need to use a few fields to map the methods to
```
Comment (danawillow) [inline products/accesscontextmanager/api.yaml]: nit: I'd replace the word 'weird' with something less negative, like 'unusual' or 'atypical'.

File: products/accesscontextmanager/api.yaml [hunk 2]
```diff
@@ -66,3 +66,200 @@ objects:
...
+      # This must be done at the provider level.
+      - !ruby/object:Api::Type::String
+        name: parent
+        input: true
+        required: true
+        description: |
+          The AccessPolicy this AccessLevel lives in.
+          Format: accessPolicies/{policy_id}
+      - !ruby/object:Api::Type::String
+        name: name
```
Comment (danawillow) [inline products/accesscontextmanager/api.yaml]: So I looked around and the redis API is another similar one to this where the name is the long-style and required. That one I handled by pretending that it's actually the short form and using a custom expander/flattener. The big advantage of that approach is consistency across resources, the disadvantage is that interpolation on it won't work directly. Do you have an opinion on whether one is better than the other for this resource?

File: products/accesscontextmanager/api.yaml [hunk 3]
```diff
@@ -66,3 +66,200 @@ objects:
...
           Time the AccessPolicy was updated in UTC.
         output: true
+  - !ruby/object:Api::Resource
+    name: 'AccessLevel'
+    # This is a weird API, so we need to use a few fields to map the methods to
+    # the right URL.
+    # create_url is the Create URL
+    # base_url is the Get and Delete and Patch URL. It is empty on purpose.
+    # List won't work yet. It should share a URL with Create.
+    create_url: |
```
[the remark this answers is not in the document]
Comment (rileykarson) [reply products/accesscontextmanager/api.yaml]: I liked them a little more aesthetically - changed to normal strings.

File: products/accesscontextmanager/api.yaml (hunk 2 above)
Comment (rileykarson) [reply products/accesscontextmanager/api.yaml]: I'm not sure what my feelings are on how to handle newer style `relative_uri` shaped `name` fields, and I figured a fully qualified name would be the most future proof until https://github.com/GoogleCloudPlatform/magic-modules/issues/937 decides on how we handle these formats consistently across products.

**Pull request merged**